velox/docs/develop/cache.rst

```diff
@@ -0,0 +1,152 @@
+
+Configuration Properties
+------------------------
+The AsyncDataCache can be enabled by setting the following config:
+
+.. code-block:: bash
+
+    async-data-cache-enabled=true
+
+
+Other Properties
+----------------
+There is a ``cache.no_retention`` session property in Velox that can be set to control if a query's cached data is retained or not after its execution.
+
+.. list-table::
+   :widths: 30 10 10 70
+   :header-rows: 1
+
+   * - Property Name
+     - Type
+     - Default Value
+     - Description
+   * - cache.no_retention
+     - bool
+     - false
+     - If set to true, evicts out a query scanned data out of in-memory cache right after the access, and also skips staging to the SSD cache.​
+
+Set the ``hive.node_scheduler_affinity`` session property accordingly to turn ON/OFF cache.no_retention.​
+
+.. code-block:: bash
+
+    SET SESSION hive.node_selection_strategy='NO_PREFERENCE'; // To turn cache.no_retention ON.​
+    SET SESSION hive.node_selection_strategy='SOFT_AFFINITY'; // To turn cache.no_retention OFF.​
+    SET SESSION hive.node_selection_strategy='HARD_AFFINITY'; // To turn cache.no_retention OFF.​
+
+
+=========
+SSD Cache
+=========
+
+Background
+----------
+The AsyncDataCache is configured to use SSD when provided.
+The SSD serves as an extension for the async data cache (file cache).
+This helps mitigate the number of reads from slower storage.
+
+Configuration Properties
+------------------------
+The SSD cache can be used by setting the following configs:
+
+.. code-block:: bash
+
+    async-data-cache-enabled=true
+    async-cache-ssd-gb=<the size of your SSD>
+    async-cache-ssd-path=<path to directory that is mounted onto SSD>
+
+.. list-table::
+   :widths: 30 10 10 70
+   :header-rows: 1
+
+   * - Property Name
+     - Type
+     - Default Value
+     - Description
+   * - async-data-cache-enabled
+     - bool
+     - true
+     - If true, enable async data cache.
+   * - async-cache-ssd-gb
+     - integer
+     - 0
+     - The size of the SSD.
+   * - async-cache-ssd-path
+     - string
+     - /mnt/flash/async_cache.
+     - The directory that is mounted onto SSD.
+
+
+Other configuration properties can also be set to control how often the async data cache writes to SSD. 
+See `Configuration Properties <../configs.rst>`_ for more SSD Cache related configuration properties.
+
+Metrics
+-------
+There are SSD cache relevant metrics that Velox emits during query execution and runtime. 
+See `Debugging Metrics <./debugging/metrics.rst>`_ and `Monitoring Metrics <../monitoring/metrics.rst>`_ for more details.
+
+
+Setup with btrfs filesystem on worker machines (Linux only)
+-----------------------------------------------------------
+NOTE: Commands below were ran successfully for worker machines of Amazon EC2 r6 instances with CentOS.
```

Comment:
We need an entry why this section is here.

Something like this maybe:
Multiple factors contribute to utilizing the SSD cache effectively. One of them is choosing the best file system that allows direct writes for best performance.

Now other filesystems allow direct writes but I remember that setting O_DIRECT needed to write with proper alignment of the memory. This isn't needed for btrfs. I don't remember the details as to why O_DIRECT didn't work with the other file systems even when it is supported.

Reply:
Changed to:

Multiple factors contribute to utilizing the SSD cache effectively. 
One of them is choosing the best file system that allows direct writes for best performance.
Btrfs was found to be a good file system to use due to its built-in data compression, 
support for O_DIRECT writes, and the ability to perform asynchronous discard operations. 
These features combine to enhance storage efficiency, improve performance, and optimize disk management.

Reply:
Please mention which version of CentOS was used. I assume CentOS9?

Reply:
Yes it was CentOS Stream 9.